### ISSUE TITLE
Inventory and tabs missing/alignment issues

### ISSUE THREAD

**Issue by mikey0000** [minetest/minetest#9788]
When starting a new game locally I have access to items/crafting etc
When starting a server the layout of the inventory is messed up and the tabs and titles are missing.

screenshot is of mineclone 2 but it does it with minetest as well.

server 5.2.0 using docker image fromhttps://github.com/linuxserver/docker-minetest
client 5.2.0

![Screenshot_20200501_211103](https://user-images.githubusercontent.com/1455933/80795400-759f2f80-8bf0-11ea-8b40-82b6ec286516.png)

Happy to help debug or even write code if need be to solve this.

I've compiled minetest server locally on Ubuntu and it behaves the same with the same options passed for the linuxserver docker image so will play around and see if it is any particular flag.

**Comment by paramat** [comment]
> screenshot is of mineclone 2 but it does it with minetest as well.

By 'minetest' do you mean Minetest Game?
We are not very familiar with MC2, so i cannot see what is wrong in that image. Please could you post a screenshot of the bug affecting Minetest Game or the Minetest Engine?

**Comment by mikey0000** [comment]
Yes it did happen in Minetest Game, I finally figured out how to fix the issue but I will get back to you with screenshots for Minetest Game, basically it has something to do with the server config vs user config, it gives me the above when I'm using a server minetest.conf, and the correct screen when I use a client minetest.conf, still trying to figure out which flag is the issue, so again will come back soon with more details and screenshots of minetest Game.

**Comment by rubenwardy** [comment]
Do you have creative enabled? The tabs won't appear in Minetest Game if creative isn't enabled

**Comment by mikey0000** [comment]
Creative was enabled in the world.mt file but not in main_config/minetest.conf

3/05/2020 4:15:13 am rubenwardy <notifications@github.com>:

**Comment by mikey0000** [comment]
Hi so back again, with screenshot, so if I have the default server minetest.conf under main_config and a set world, doesn't matter if I run minetest --server or minetestserver

I get the following screen
![Screenshot_20200503_204530](https://user-images.githubusercontent.com/1455933/80909849-87630d00-8d7f-11ea-89cc-87fa6968771b.png)

the world.mt file has the following in it
`enable_damage = false
creative_mode = true
auth_backend = sqlite3
player_backend = sqlite3
backend = sqlite3
gameid = minetest`

and I am following the wiki 
https://wiki.minetest.net/Setting_up_a_server/Debian

only if I enable creative_mode in the server config does the screen then change to 
![Screenshot_20200503_205151](https://user-images.githubusercontent.com/1455933/80909886-fc364700-8d7f-11ea-809c-3873d8a1748b.png)

Both following the wiki and as a new user I'd expect to see the tabs, even if they were empty.
I'd also expect that enabling settings specifically for a world (within the world.mt) would work for that world.

on a side note is it possible to run a multi-world server?
Thanks

**Comment by sfan5** [comment]
> I'd expect to see the tabs, even if they were empty.

That wouldn't be very useful in my opinion. But either way this is not the behaviour, tabs don't appear in survival.

> I'd also expect that enabling settings specifically for a world (within the world.mt) would work for that world

That should work yes. The might be a problem when you have `creative_mode = false` in minetest.conf and `creative_mode = true` in world.mt but I don't know how that is handled.

> on a side note is it possible to run a multi-world server?

It is not. The way people usually do this is by running multiple instance of minetestserver on different ports.

**Comment by rubenwardy** [comment]
There's a separate issue here for the dedicated server not taking settings from world.mt

> The might be a problem when you have creative_mode = false in minetest.conf and creative_mode = true in world.mt but I don't know how that is handled.

The main menu sets the minetest.conf using the world.mt. This is not the correct location, given that the server doesn't have a main menu

**Comment by rubenwardy** [comment]
> the layout of the inventory is messed up 

What do you mean by this?

**Comment by mikey0000** [comment]
OK, thanks for all the replies, I can confirm creative_mode was not set in minetest.conf.

Now that I have a better understanding of how things work in minetest and minetest game. All my other comments are no longer relevant. 

The core issue is minetestserver not picking up creative_mode in the world.mt.

**Comment by CodePhase** [comment]
I have just upgraded my minetestserver to version 5.3.0 and minetest_game and it broke my 'enable_damage = false' setting in both my world .conf and world.mt (it's set in both places).  It seems to be completely ignored in world.mt so I just commented it out in that file.  When the .conf file has 'enable_damage = true' the health bar is displayed.  When 'enable_damage = false' is set the health bar is not displayed and players takes damage and die from falling, mob attacks, and lava, but do not drown.

**Comment by sfan5** [comment]
> When 'enable_damage = false' is set the health bar is not displayed and players take damage and die from falling, mob attacks, and lava, but do not drown.

That sounds like #9022, which is not an engine issue and doesn't have anything to do with this.

**Comment by CodePhase** [comment]
@sfan5, confirmed.  The issue disappears when I disable the 3d_armor mod.  Thank you!
